govcd/orgvdcnetwork.go

```diff
@@ -236,3 +236,111 @@ func (vdc *Vdc) FindEdgeGatewayNameByNetwork(networkName string) (string, error)
 	}
 	return "", ErrorEntityNotFound
 }
+
+// getParentVdc retrieves the VDC to which the network is attached
+func (orgVdcNet *OrgVDCNetwork) getParentVdc() (*Vdc, error) {
+	for _, link := range orgVdcNet.OrgVDCNetwork.Link {
+		if link.Type == "application/vnd.vmware.vcloud.vdc+xml" {
+
+			vdc := NewVdc(orgVdcNet.client)
+
+			_, err := orgVdcNet.client.ExecuteRequest(link.HREF, http.MethodGet,
+				"", "error retrieving parent vdc: %s", nil, vdc.Vdc)
+			if err != nil {
+				return nil, err
+			}
+
+			return vdc, nil
+		}
+	}
+	return nil, fmt.Errorf("could not find a parent Vdc for network %s", orgVdcNet.OrgVDCNetwork.Name)
+}
+
+// getEdgeGateway retrieves the edge gateway connected to a routed network
+func (orgVdcNet *OrgVDCNetwork) getEdgeGateway() (*EdgeGateway, error) {
+	// If it is not routed, returns an error
+	if orgVdcNet.OrgVDCNetwork.Configuration.FenceMode != types.FenceModeNAT {
+		return nil, fmt.Errorf("network %s is not routed", orgVdcNet.OrgVDCNetwork.Name)
+	}
+	vdc, err := orgVdcNet.getParentVdc()
+	if err != nil {
+		return nil, err
+	}
+
+	// Since this function can be called from Update(), we must take into account the
+	// possibility of a name change. If this is happening, we need to retrieve the original
+	// name, which is still stored in the VDC.
+	oldNetwork, err := vdc.GetOrgVdcNetworkById(orgVdcNet.OrgVDCNetwork.ID, false)
+	if err != nil {
+		return nil, err
+	}
+	networkName := oldNetwork.OrgVDCNetwork.Name
+
+	edgeGatewayName, err := vdc.FindEdgeGatewayNameByNetwork(networkName)
+	if err != nil {
+		return nil, err
+	}
+
+	return vdc.GetEdgeGatewayByName(edgeGatewayName, false)
+}
+
+// UpdateAsync will change the contents of a network using the information in the
+// receiver data structure.
+func (orgVdcNet *OrgVDCNetwork) UpdateAsync() (Task, error) {
+	if orgVdcNet.OrgVDCNetwork.HREF == "" {
+		return Task{}, fmt.Errorf("cannot update network: HREF is empty")
```

Comment:
We should update all errors to be more specific-> "cannot update Org VDC network: HREF is empty"

Reply:
updated